discord-bot/bot/extensions/tasks.py

```diff
@@ -0,0 +1,304 @@
+        task="Please explain the moon landing to a six year old.",
+        timeout=MAX_TASK_TIME,
+    )
+    resp = await ctx.respond(
+        "Task - Respond to the prompt as if you were the Assistant:",
+        flags=hikari.MessageFlag.EPHEMERAL,
+        components=view,
+    )
+    await view.start(await resp.message())
+
+
+class RatingView(miru.View):
+    """View for rating a task."""
+
+    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.presses: list[str] = []
+
+    def _close_if_all_pressed(self) -> None:
+        if len(self.presses) == 5:
+            self.stop()
+
+    @miru.button(label="1", style=hikari.ButtonStyle.PRIMARY)
+    async def button_1(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        if button.label not in self.presses:
+            self.presses.append("1")
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+        self._close_if_all_pressed()
+
+    @miru.button(label="2", style=hikari.ButtonStyle.PRIMARY)
+    async def button_2(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        if button.label not in self.presses:
+            self.presses.append("2")
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+        self._close_if_all_pressed()
+
+    @miru.button(label="3", style=hikari.ButtonStyle.PRIMARY)
+    async def button_3(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        if button.label not in self.presses:
+            self.presses.append("3")
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+        self._close_if_all_pressed()
+
+    @miru.button(label="4", style=hikari.ButtonStyle.PRIMARY)
+    async def button_4(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        if button.label not in self.presses:
+            self.presses.append("4")
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+        self._close_if_all_pressed()
+
+    @miru.button(label="5", style=hikari.ButtonStyle.PRIMARY)
+    async def button_5(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        if button.label not in self.presses:
+            self.presses.append("5")
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+        self._close_if_all_pressed()
+
+    @miru.button(label="Reset", style=hikari.ButtonStyle.DANGER)
+    async def reset_button(self, button: miru.Button, ctx: miru.ViewContext) -> None:
+        self.presses = []
+        await ctx.respond(f"Received response: {button.label}", flags=hikari.MessageFlag.EPHEMERAL)
+
+
+class SelectRating(miru.View):
+    """View for rating a task with a select menu."""
+
+    @miru.select(
+        options=[
+            hikari.SelectMenuOption(
```

Comment:
There is `scale: RatingScale` that defines min and max for the rating.

Reply:
the `tasks.py` file was for testing different types of user input. If we decide to go with select menus for rating, I will update the code in `work.py`. I'll rename `task.py` to `task_tests.py` to make that more clear.